src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java

```diff
@@ -38,7 +38,23 @@ public CustomerDto findById(final Long customerId) {
     @Override
     @Transactional
     public CustomerDto save(CustomerDto customerDto) {
-        Customer customer = Optional.ofNullable(customerDto.getId())
+        Customer customer = buildCustomerEntity(customerDto);
```

Comment:
hm, whats wrong with original implementation?

Reply:
I told about it at daily meeting. We shouldn't change registration date and create admin when updating customer... to avoid if .. else and make code readability I separated logic to two methods